experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+            except AssertionError:
+                errmsg = """rxfreq must be a number (kHz) between min and max frequencies for the 
+                            radar"""
+                raise ExperimentException(errmsg)
+
+        else:  # TX-specific mode , without a clear frequency search.
+            # In this mode, txfreq is required along with the other requirements.
+            try:
+                assert isinstance(exp_slice['txfreq'], int) or isinstance(exp_slice['txfreq'],
+                                                                          float)
+                assert (exp_slice['txfreq'] * 1000) < self.options.max_freq
+                assert (exp_slice['txfreq'] * 1000) > self.options.min_freq
+            except AssertionError:
+                errmsg = """txfreq must be a number (kHz) between min and max frequencies for the 
+                                            radar"""
+                raise ExperimentException(errmsg)
+            for freq_range in self.options.restricted_ranges:
+                try:
+                    assert exp_slice['txfreq'] not in range(freq_range[0], freq_range[1])
+                except AssertionError:
+                    errmsg = """txfreq is within a restricted frequency range"""
+                    raise ExperimentException(errmsg)
+                    # TODO or could just log this and set to default frequency, or the closest
+                    # frequency that is available. Do we want to remove default frequency?
+
+    def set_slice_defaults(self, exp_slice):
+        """
+        Set up defaults in case of some parameters being left blank.
+        :param exp_slice: 
+        :return: slice_with_defaults: updated slice
+        """
+
+        slice_with_defaults = copy.deepcopy(exp_slice)
+
+        if 'txantennas' not in exp_slice:
+            slice_with_defaults['txantennas'] = [i for i in range(0,
+                                                                  self.options.main_antenna_count)]
+            # all possible antennas.
+        if 'rx_main_antennas' not in exp_slice:
+            slice_with_defaults['rx_main_antennas'] = [i for i in
+                                                       range(0, self.options.main_antenna_count)]
+        if 'rx_int_antennas' not in exp_slice:
+            slice_with_defaults['rx_int_antennas'] = \
+                [i for i in range(0, self.options.interferometer_antenna_count)]
+        if 'pulse_shift' not in exp_slice:
+            slice_with_defaults['pulse_shift'] = [0 for i in range(0, len(
+                slice_with_defaults['pulse_sequence']))]
+        if 'scanboundflag' and 'scanbound' not in exp_slice:
+            slice_with_defaults['scanboundflag'] = False
+            slice_with_defaults['scanbound'] = None
+        elif 'scanboundflag' not in exp_slice:  # but scanbound is
+            slice_with_defaults['scanboundflag'] = True
```

Comment:
Should we tell the user here that they should have specified scanboundflag is True?